htlcswitch/link.go

```diff
@@ -1306,6 +1307,20 @@ func (l *channelLink) htlcManager() {
 		case <-l.quit:
 			return
 		}
+
+		// After we are finished processing the event, if the link is
+		// flushing, we check if the channel is clean and invoke the
+		// post-flush hook if it is.
+		if l.IsFlushing() && l.channel.IsChannelClean() {
+			// This will not block since flushCont must be full.
+			// We Read instead of Take to ensure a new flush
+			// operation can't be initiated until the continuation
+			// for the current flush has completed.
+			l.flushCont.Read()()
+
+			// Reset the flushCont MVar. This will also not block.
+			l.flushCont.Take()
```

Comment:
Ah I see here now that the state can basically be fully re-used. Eg: dyn commitments when we want to do a series of independent updates w/o actually tearing down the link.